syntax = "proto3";

package messenger;

// Authentication service
service AuthService {
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc ValidateToken(ValidateTokenRequest) returns (ValidateTokenResponse);
}

// Login request
message LoginRequest {
  string username = 1;
  string password = 2;
}

// Login response
message LoginResponse {
  bool success = 1;
  string token = 2;
  string message = 3;
  int32 user_id = 4;
  string username = 5;
}

// Token validation request
message ValidateTokenRequest {
  string token = 1;
}

// Token validation response
message ValidateTokenResponse {
  bool valid = 1;
  int32 user_id = 2;
  string username = 3;
}